Calculator/Calculator/Model/Enum/ExpressionParser.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  ExpressionParser.swift
+//  Calculator
+//
+//  Created by Min Hyun on 2023/05/31.
+//
+
+enum ExpressionParser {
+    static func parse(from input: String) -> Formula {
+        var operatorsQueue = CalculatorItemQueue<Operator>()
+        var operandsQueue = CalculatorItemQueue<Double>()
+
+        let operandComponents = self.componentsByOperators(from: input).compactMap { Double($0) }
+        let operatorComponents = input.compactMap { Operator(rawValue: $0) }
+        
+        operandComponents.forEach { operandsQueue.enqueue($0) }
+        operatorComponents.forEach { operatorsQueue.enqueue($0) }
+
+        let result = Formula(operands:operandsQueue, operators: operatorsQueue)
+
+        return result
+    }
+        
+    static private func componentsByOperators(from input: String) -> [String] {
+        let result = Operator.allCases.reduce([input]) { resultArray, operatorItem in
+            resultArray.map { $0.split(with:operatorItem.rawValue) }.flatMap { $0 }
+        }
+        
+        return result
+    }
```

Comment:
```suggestion
        return Formula(operands:operandsQueue, operators: operatorsQueue)
    }
        
    static private func componentsByOperators(from input: String) -> [String] {
        return Operator.allCases.reduce([input]) { resultArray, operatorItem in
            resultArray.map { $0.split(with:operatorItem.rawValue) }.flatMap { $0 }
        }
    }
```

이러면 변수의 사용을 줄일 수 있겠군요!

Reply:
오! 리턴에서 바로 호출해도 되겠네요! 수정해보겠습니다~